CrazyCanvas/Source/GUI/LobbyGUI.cpp

```diff
@@ -216,6 +231,62 @@ void LobbyGUI::OnButtonJoinClick(Noesis::BaseComponent* pSender, const Noesis::R
 	}
 }
 
+void LobbyGUI::StartSelectedServer(Noesis::Grid* pGrid)
+{
+	if (!m_Servers.empty())
+	{
+		for (auto& server : m_Servers)
+		{
+			if (server.second.ServerGrid == pGrid)
+			{
+				LambdaEngine::GUIApplication::SetView(nullptr);
+
+				SetRenderStagesActive();
+
+				State* pPlaySessionState = DBG_NEW PlaySessionState(server.second.EndPoint.GetAddress());
+				StateManager::GetInstance()->EnqueueStateTransition(pPlaySessionState, STATE_TRANSITION::POP_AND_PUSH);
+			}
+		}
+	}
+}
+
+bool LobbyGUI::CheckServerStatus()
+{
+	TArray<uint64> serversToRemove;
+
+	for (auto& server : m_Servers)
+	{
+		Timestamp deltaTime = EngineLoop::GetTimeSinceStart() - server.second.LastUpdate;
```

Comment:
Could this be moved outside of the loop?

Reply:
You mean just the declaration of deltaTime? If so, then yes it could be moved outside.

Reply:
No, I meant it's unnecessary to fetch the time in each iteration of the loop. DeltaTime needs to be greater than 5 seconds for something to happen. Each iteration of the loop is probably not going to be more than a couple of microseconds, so fetching the time in each iteration isn't meaningful.

Reply:
I see what you meant now, sorry I misunderstood the first time